File: staGledas-UI/stagledas_mobile/lib/models/klub_filmova.dart
import 'package:json_annotation/json_annotation.dart';
import 'korisnik.dart';
import 'klub_filmova_clanovi.dart';

part 'klub_filmova.g.dart';

@JsonSerializable()
class KlubFilmova {
  int? id;
  String? naziv;
  String? opis;
  String? slika;
  int? vlasnikId;
  bool? isPrivate;
  int? maxClanova;
  DateTime? datumKreiranja;
  int? brojClanova;
  Korisnik? vlasnik;
  List<KlubFilmovaClanovi>? clanovi;

  KlubFilmova({
    this.id,
    this.naziv,
    this.opis,
    this.slika,
    this.vlasnikId,
    this.isPrivate,
    this.maxClanova,
    this.datumKreiranja,
    this.brojClanova,
    this.vlasnik,
    this.clanovi,
  });

  factory KlubFilmova.fromJson(Map<String, dynamic> json) =>
      _$KlubFilmovaFromJson(json);

  Map<String, dynamic> toJson() => _$KlubFilmovaToJson(this);
}

File: staGledas.Model/Models/KlubFilmova.cs
using System;
using System.Collections.Generic;

namespace staGledas.Model.Models
{
    public class KlubFilmova
    {
        public int Id { get; set; }
        public string? Naziv { get; set; }
        public string? Opis { get; set; }
        public byte[]? Slika { get; set; }
        public int VlasnikId { get; set; }
        public bool IsPrivate { get; set; }
        public int? MaxClanova { get; set; }
        public DateTime DatumKreiranja { get; set; }
        public int BrojClanova { get; set; }
        public virtual Korisnici? Vlasnik { get; set; }
        public virtual ICollection<KlubFilmovaClanovi>? Clanovi { get; set; }
    }
}
